### DISCUSSION TITLE
Community triggers backward compatibility

### DISCUSSION THREAD

**Discussion by serdarkalayci** [kedacore/keda#3091] [Q&A / Need Help] [answered]
As far as I can see from the code, trigger specific fields are inside this map
Metadata map[string]string `json:"metadata"` 
When a maintainer of a community scaler does a breaking change, and it gets released with the operator's next release, there's no way to enforce the users of the older version to update their Scaler manifests to update right? They have to see it from the operator logs, that specific scaler requires a new field to work and update that manifest manually. This way, the probability of CRDs to change is very low, but it's up to KEDA's users to check if their scalers still work or not after an operator upgrade.
Am I assuming wrong?

**Comment by tomkerkhove** [comment, marked as answer]
Built-in scalers have to be backwards compatible unless we bump the major version so that should not be a problem.

**Reply by zroubalik** [reply]
And in case there's a change, the deprecation is long process. Release notes should also help with this.

**Reply by serdarkalayci** [reply]
In fact I'm asking on behalf of another similar open source project External Secrets Operator (https://github.com/external-secrets/external-secrets) which has a similar problem (community developed secret sources) and those secret sources are prone to force a breaking change more than scalers here (public cloud providers tend to change the ways to access their services more). We were discussing different approaches to reduce the impact on our users, and decided to learn how you people approach the same problem.

**Reply by tomkerkhove** [reply]
That's a fair question but you can rest assured - We won't allow breaking changes in scalers!

**Reply by zroubalik** [reply]
@tomkerkhove I wouldn't be that stritct though 😃  I think that if the change is well communicated and **slow** - ie for a period of time you support both old and new version (and notifying users to do the migration - in the logs atm, but sending more events in the future), then it is fine.

The ScaledObject CRD which hosts the scalers definition is still `v1alpha1` and this by it's nature can be changed.

But again, we should be really careful.

**Reply by serdarkalayci** [reply]
Thanks for the answers. I'll discuss these with the rest of the team. As we gain more and more interest from the community lately, we decided to take these subjects at hand before they get out of control. Your input means a lot.

**Reply by tomkerkhove** [reply]
> @tomkerkhove I wouldn't be that stritct though 😃 I think that if the change is well communicated and **slow** - ie for a period of time you support both old and new version (and notifying users to do the migration - in the logs atm, but sending more events in the future), then it is fine.

Definately, as long as old & new both still work.
